fedimint-api/src/tiered_multi.rs

```diff
@@ -141,19 +142,42 @@ where
 }
 
 impl TieredMulti<()> {
-    // TODO: move somewhere else?
-    pub fn represent_amount<K>(mut amount: Amount, tiers: &Tiered<K>) -> TieredMulti<()> {
-        let coins = tiers
-            .tiers()
-            .rev()
-            .map(|&amount_tier| {
-                let res = amount / amount_tier;
-                amount %= amount_tier;
-                (amount_tier, vec![(); res as usize])
-            })
-            .collect();
+    /// Determines the denominations to use when representing an amount
+    ///
+    /// Algorithm tries to leave the user with a target number of `denomination_sets` starting
+    /// at the lowest denomination.  `self` gives the denominations that the user already has.
+    pub fn represent_amount<K, V>(
+        mut amount: Amount,
+        current_denominations: &TieredMulti<V>,
+        tiers: &Tiered<K>,
+        denomination_sets: u16,
+    ) -> TieredMulti<()> {
+        let mut denominations: Vec<(Amount, ())> = vec![];
+        let mut tiers: Vec<Amount> = tiers.tiers().cloned().collect();
+        tiers.sort();
+
+        // try to hit the target `denomination_sets`
+        for tier in tiers.clone() {
```

Comment:
Even if you want to stick to the `Vec` you could use `&tiers` or `tiers.iter()` to not have to clone here. But just going with `for &tier in tiers.tiers() {` is probably best (`Amount` is `Copy`, so you can just dereference to get the owned value without cloning).